File: unimonpark/src/main/java/co/edu/unimonserrate/unimonpark/entity/Membresia.java
package co.edu.unimonserrate.unimonpark.entity;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "membresias")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor
public class Membresia {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idMembresia;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_vehiculo", nullable = false)
    private Vehiculo vehiculo;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_tarifa", nullable = false)
    private Tarifa tarifa;

    @Column(name = "fecha_inicio", nullable = false)
    private LocalDateTime fechaInicio;

    @Column(name = "fecha_fin", nullable = false)
    private LocalDateTime fechaFin;

    @Column(name = "monto_pagado", nullable = false)
    private BigDecimal montoPagado;

    @Column(name = "fecha_creacion", nullable = false)
    private LocalDateTime fechaCreacion;
}

File: unimonpark/unimonpark-frontend/src/types/membresia.ts
export interface Membresia {
  idMembresia: number;
  idVehiculo: number;
  placaVehiculo: string;
  idTarifa: number;
  nombreTarifa: string;
  fechaInicio: string;
  fechaFin: string;
  montoPagado: number;
  activa: boolean;
}

export type MembresiaPayload = {
  idVehiculo: number;
  idTarifa: number;
};